parkinglot/model/ParkingLot.java

```diff
@@ -0,0 +1,139 @@
+package Designs.parkinglot.model;
+
+import Designs.parkinglot.parkingstrategy.ParkingStrategy;
+
+import java.util.*;
+
+public class ParkingLot {
+    private final String parkingLotId;
+    public static final Integer RESERVED_SLOT_FOR_TRUCK = 1;
+    public static final List<Integer> RESERVED_SLOT_FOR_BIKE = List.of(2, 3);
+    private final Integer noOfFloors;
+    private final Integer noOfSlotsPerFloor;
+    private final List<Floor> floors = new ArrayList<>();
+    // Used strategy pattern here
+    private final ParkingStrategy parkingStrategy;
+    // Maintaining reference of available floor-slots by vehicle type to get next available slot faster and for display operations
+    private final Map<VehicleType, TreeMap<Integer, TreeSet<Integer>>> availableFloorSlotMapByVehicleType = new HashMap<>();
+    // Maintaining reference of occupied floor-slots by vehicle type for display operations
+    private final Map<VehicleType, TreeMap<Integer, TreeSet<Integer>>> occupiedFloorSlotMapByVehicleType = new HashMap<>();
+
+    public ParkingLot(String parkingLotId, Integer noOfFloors, Integer noOfSlotsPerFloor, ParkingStrategy parkingStrategy) {
+        this.parkingLotId = parkingLotId;
+        this.parkingStrategy = parkingStrategy;
+        this.noOfFloors = noOfFloors;
+        this.noOfSlotsPerFloor = noOfSlotsPerFloor;
+        initialiseParkingLot(parkingLotId, noOfFloors, noOfSlotsPerFloor);
+    }
+
+    private void initialiseParkingLot(String parkingLotId, Integer noOfFloors, Integer noOfSlotsPerFloor) {
+        for (int floorNum = 1; floorNum <= noOfFloors; floorNum++) {
+            List<Slot> slots = new ArrayList<>();
+            for (int slotNum = 1; slotNum <= noOfSlotsPerFloor; slotNum++) {
+                Slot slot = new Slot(slotNum, floorNum, parkingLotId, null, Slot.Status.VACATED);
+                if (slotNum == RESERVED_SLOT_FOR_TRUCK) {
+                    slot.setVehicleType(VehicleType.TRUCK);
```

Comment:
Extract function. Same comment for all the if-else handling. Make the initialise function easy to understand by making it smaller.

Reply:
Noted. Will update it. Thanks for reviewing.